odrivemotor/src/main.py

```diff
@@ -18,7 +18,4 @@ async def main(address: str):
     await module.start()
 
 if __name__ == "__main__":
-    if len(sys.argv) != 2:
-        raise Exception("Need socket path as command line argument")
-
     asyncio.run(main(sys.argv[1]))
```

Comment:
I'm pretty sure doing this by argument order is fragile and that there should be a way to get the socket path from the command line args without doing by location

Reply:
would this work?

Reply:
I don't think we need this anymore because the modmanager raises an error if the socket path isn't provided now - https://docs.viam.com/use-cases/create-module/#write-an-entry-point-main-program-file. You don't need to pass in an argument to main.

It used to not do that, and this module is one of the first we wrote ever so I think this is a case of code that's no longer necessary

Reply:
Your main function doesn't consume this anymore.

Reply:
or shouldn't?

Reply:
![Screenshot 2024-08-01 at 14 34 52](https://github.com/user-attachments/assets/90153422-554e-43a9-9af6-69d056a21f0c)

Reply:
yeah the `from_args` helper was what I was thinking of

Reply:
I'm not familiar with `from_args`... so to clarify, we don't need these lines at all? or they need to be replaced using `from _args`?

Reply:
We need to replace your main function with what is in the docs. If you read the viam python SDK link - it explains how from args is doing what you are doing and passing it to the main.

This is a function that didn't exist when you first wrote this, but it is now the way we recommend writing the Python module's main functions. The docs have been updated to reflect this.

Reply:
I can't find the explanation of `from_args`, but I think I understand now how it replaces using `address`. just added that change

Reply:
I don't have an odrive to test tho, but I think it should be fine. let me know if there's any issues @raybjork

Reply:
> You shouldnt need odrive hardware to test this change since its only concerned with initialization of the module
We always test :) but this should be totally fine. If there's a problem with the new version, we will know immediately. @martha-johnston, the link I attached to my previous comment links to the Python SDK docs and that function directly.

Reply:
ah I was looking at the first link I see it now. and yeah just let me know if there's any issues when I upload the new version

Reply:
issues with the hardware that is